src/EventListener/DataContainer/TagListener.php

```diff
@@ -75,4 +76,108 @@ private function getManager(string $alias): ManagerInterface
     {
         return $this->registry->get($alias);
     }
+
+    /**
+     * Auto-generate the tag alias if it has not been set yet
+     *
+     * @param mixed         $value
+     * @param DataContainer $dc
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    public function generateAlias($value, DataContainer $dc)
```

Comment:
This whole method needs to be reworked to use strict types and comparisons, database **service**, etc.